app/src/main/java/com/hedvig/app/feature/offer/OfferViewModel.kt

```diff
@@ -48,16 +53,36 @@ abstract class OfferViewModel : ViewModel() {
             val checkoutParameter: CheckoutParameter
         ) : Event()
 
-        object ApproveError : Event()
+        data class ApproveError(
+            val postSignScreen: PostSignScreen,
+        ) : Event()
+
         data class ApproveSuccessful(
-            val moveDate: LocalDate?
+            val startDate: LocalDate?,
+            val postSignScreen: PostSignScreen,
+            val bundleDisplayName: String,
         ) : Event()
 
         data class StartSwedishBankIdSign(val quoteIds: List<String>, val autoStartToken: String) : Event()
 
         object DiscardOffer : Event()
     }
 
+    enum class PostSignScreen {
```

Comment:
Would it make more sense to have this enum in `GetPostSignDependenciesUseCase`? That would decouple network types from the viewmodel.

Reply:
Sure!